drivers/spi/spi_esp32_spim.c

```diff
@@ -330,13 +330,18 @@ static int IRAM_ATTR spi_esp32_configure(const struct device *dev,
 	data->trans_config.line_mode.cmd_lines = 1;
 
 	/* SPI mode */
-	hal_dev->mode = 0;
 	if (SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPOL) {
-		hal_dev->mode = BIT(0);
-	}
-
-	if (SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPHA) {
-		hal_dev->mode |= BIT(1);
+		if (SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPHA) {
+			hal_dev->mode = 3;
+		} else {
+			hal_dev->mode = 2;
+		}
+	} else {
+		if (SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPHA) {
```

Comment:
I like the idea of aligning with the way the information is typically presented in the ESP-IDF documentation (i.e. "modes" 0-3 rather than actually talking about CPOL/CPHA. 
That being said, there's also the option of fixing the original code by "just" swapping BIT(0) and BIT(1), which was the original bug, right?

```
	if (SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPHA) {
		hal_dev->mode = BIT(0);
	}

	if (SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPOL) {
		hal_dev->mode |= BIT(1);
	}
```

Reply:
Yes, you are right. I am open to either possibilities.

Reply:
@sylvioalves can probably suggest what would his preference be